pypeit/spectrographs/mmt_bluechannel.py

```diff
@@ -0,0 +1,292 @@
+        par['reduce']['skysub']['global_sky_std']  = False
+
+        # cosmic ray rejection parameters for science frames
+        par['scienceframe']['process']['sigclip'] = 5.0
+        par['scienceframe']['process']['objlim'] = 2.0
+
+        # Set the default exposure time ranges for the frame typing
+        par['calibrations']['standardframe']['exprng'] = [None, 600]
+        par['calibrations']['arcframe']['exprng'] = [10, None]
+        par['calibrations']['darkframe']['exprng'] = [300, None]
+        par['scienceframe']['exprng'] = [10, None]
+
+        # Need to specify this for long-slit data
+        par['calibrations']['slitedges']['sync_predict'] = 'nearest'
+
+        # Sensitivity function parameters
+        par['sensfunc']['polyorder'] = 7
+        par['sensfunc']['IR']['telgridfile'] = resource_filename('pypeit', '/data/telluric/TelFit_Paranal_VIS_4900_11100_R25000.fits')
+
+        return par
+
+    def bpm(self, filename, det, shape=None, msbias=None):
+        """
+        Generate a BPM
+
+        Parameters
+        ----------
+        det : int, REQUIRED
+        **null_kwargs:
+           Captured and never used
+
+        Returns
+        -------
+        badpix : ndarray
+
+        """
+        # Get the empty bpm: force is always True
+        bpm_img = self.empty_bpm(filename, det, shape=shape)
+
+        # Fill in bad pixels if a master bias frame is provided
+        if msbias is not None:
+            return self.bpm_frombias(msbias, det, bpm_img)
+
+        if det == 1:
+            msgs.info("Using hard-coded BPM for  Blue Channel")
+
+            bpm_img[-1, :] = 1
+
+        else:
+            msgs.error(f"Invalid detector number, {det}, for MMT Blue Channel (only one detector).")
+
+        return bpm_img
+
+    def configuration_keys(self):
+        return ['dispname']
+
+    def check_frame_type(self, ftype, fitstbl, exprng=None):
+        """
+        Check for frames of the provided type.
+        """
+        good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
+        if ftype == 'bias':
+            return fitstbl['idname'] == 'zero'
+        if ftype == 'science':
+            return good_exp & (fitstbl['lampstat01'] == 'off') & (fitstbl['idname'] == 'object')
+        if ftype == 'standard':
+            return good_exp & (fitstbl['lampstat01'] == 'off') & (fitstbl['idname'] == 'object')
+        if ftype in ['arc', 'tilt']:
+            # should flesh this out to include all valid arc lamp combos
+            return good_exp & (fitstbl['lampstat01'] != 'off') & (fitstbl['idname'] == 'comp') & (fitstbl['decker'] != '5.0x180')
+        if ftype in ['trace']:
+            # i think the bright lamp, BC, is the only one ever used for this
+            return good_exp & (fitstbl['lampstat01'] == 'BC')
+        if ftype in ['pixelflat', 'illumflat']:
+            # imagetyp should always be set to flat for these
+            return good_exp & (fitstbl['idname'] == 'flat')
+
```

Comment:
In most cases, we use the same frames for tracing and flat fielding. Make sure you do want to use different exposures for trace and pixel flat.

Reply:
yeah, slit tracing needs more calibration data to test. all the data i have currently uses the 180" slit which is bigger than the detector. will get some data with shorter slits to cover that. also, i don't actually want to force `idname` to `flat` for `illumflat` because science data can be used as sky flats in many cases. more accurately, sky flats are usually taken while tracking so the `IMAGETYP` header gets set to `object` unless the user remembers to change it. i'll go and change that.